### ISSUE TITLE
Title component inconsistent prop defaults

### ISSUE THREAD

**Issue by seanforyou23** [patternfly/patternfly-react#1439]
`size` prop is required without a default
`headingLevel` is not required, but has a default

We should align these two. `headingLevel` should be required as a default of h1 leads to non-semantic markup structures for users who go with defaults. It's not good for accessibility as it stands out of the box.

Further, we'd probably benefit from shortening the name from `headingLevel` to simply `level`.

**Comment by rachael-phillips** [comment]
After discussion, we'd like to move forward with making the headinglevel required but not the name change.

**Comment by tlabaj** [comment]
closed by #3922
